### DISCUSSION TITLE
Is it possible to listen on non-loopback address in Windows?

### DISCUSSION THREAD

**Discussion by ghost** [DNSCrypt/dnscrypt-proxy#2584] [Q&A]
Is it? So far the only "listening on" address range I can use is 127.0.0.0-127.255.255.255 without DNSCrypt Proxy failing to start. I need it to listen on non-loopback address.

**Comment by lifenjoiner** [comment]
Definitely yes.
You can listen to any of your IP in scope of: loopback, link-local, LAN, WAN, or local.
Loopback is the safe one recommended for local use.

**Comment by ghost** [comment]
It doesn't though. It refuses to start when "Listen To" address is anything other than 0.0.0.0-0.255.255.255 range or 127.0.0.0-127.255.255.255 (loopback) range.

I set "Listen To" address to loopback and use loopback only for non-VPN interface DNS address, but DNSCrypt-Proxy continues to resolve addresses for VPN interace, which is not set to use loopback. 

Basically, DNSCrypt uses loopback and only loopback and forces itself onto interfaces that do not use loopback for DNS address...

**Comment by lifenjoiner** [comment]
You can only listen to your own IP, not arbitrary. Make it clear what IP of the adapter you have first.

**Comment by ghost** [comment]
For example, if I set DNSCrypt-Porxy to listen on 192.168.1.1, then its log records error about BIND and CONTEXT.
